torch/csrc/jit/script/init.cpp

```diff
@@ -748,11 +748,20 @@ void initJitScriptBindings(PyObject* module) {
             return bool(self.find_method(name));
           })
       .def(
-          "_method_names", [](Object& self) {
+          "_method_names",
+          [](Object& self) {
             return fmap(self.get_methods(), [](const Method& method) {
               return method.name();
             });
-          });
+          })
+      .def("__str__", [](Object& self) {
+        if (auto method = self.find_method("__str__")) {
+          Stack s;
+          method->run(s);
+          return s.back().toString()->string();
+        }
+        throw std::runtime_error("__str__ is not defined!");
```

Comment:
It's somewhat niche, but in theory you should be raising `NotImplemented` error here. For example, in `a + b` if `__add__` isn't defined on `a` `__radd__` is called on b. Throwing something other than `NotImplemented`  in `__add__` will prevent this behavior. I don't if there are any other places where there's a similar hierarchy with `__str__`.